### PR TITLE
Integration test fixes

### PR DESCRIPTION
**Pull request by CMogilko**
* Tests do not depend on the same tables no more - no failures due parallel modifications
* Fixed wrong asserts
* Added go 1.10 & 1.11 to CI
* Internal util conn wrapper to test socket leaks

### CODE REVIEW

File: internal/integration/reql_tests/reql_aggregation_test.go [hunk 1]
```diff
@@ -1219,11 +1219,11 @@ func (suite *AggregationSuite) TestCases() {
...
...e{}{45, 1}, []interface{}{46, 1}, []interface{}{47, 1}, []interface{}{48, 1}, []interface{}{49, 1}, []interface{}{50, 1}, []interface{}{51, 1}, []interface{}{52, 1}, []interface{}{53, 1}, []interface{}{54, 1}, []interface{}{55, 1}, []interface{}{56, 1}, []interface{}{57, 1}, []interface{}{58, 1}, []interface{}{59, 1}, []interface{}{60, 1}, []interface{}{61, 1}, []interface{}{62, 1}, []interface{}{63, 1}, []interface{}{64, 1}, []interface{}{65, 1}, []interface{}{66, 1}, []interface{}{67, 1}, []interface{}{68, 1}, []interface{}{69, 1}, []interface{}{70, 1}, []interface{}{71, 1}, []interface{}{72, 1}, []interface{}{73, 1}, []interface{}{74, 1}, []interface{}{75, 1}, []interface{}{76, 1}, []interface{}{77, 1}, []interface{}{78, 1}, []interface{}{79, 1}, []interface{}{80, 1}, []interface{}{81, 1}, []interface{}{82, 1}, []interface{}{83, 1}, []interface{}{84, 1}, []interface{}{85, 1}, []interface{}{86, 1}, []interface{}{87, 1}, []interface{}{88, 1}, []interface{}{89, 1}, []interface{}{90, 1}, []interface{}{91, 1}, []interface{}{92, 1}, []interface{}{93, 1}, []interface{}{94, 1}, []interface{}{95, 1}, []interface{}{96, 1}, []interface{}{97, 1}, []interface{}{98, 1}, []interface{}{99, 1}}}
```
Comment (vitalyisaev2) [inline internal/integration/reql_tests/reql_aggregation_test.go:1221]: Is it a generated code? Or the original author made it by hand?

File: internal/utils/conn_counter.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	"sync/atomic"
+	"time"
+)
+
+var connsCount int64
+var printer sync.Once
+
+var mu sync.Mutex
+var createdStacks = make(map[string]int)
+var reg = regexp.MustCompile(`0[xX][0-9a-fA-F]+`)
```
Comment (vitalyisaev2) [inline internal/utils/conn_counter.go:21]: Perhaps these global variables would look better if you encapsulate them into a single structure

File: internal/utils/conn_counter.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+// Socket leak debug net.Conn wrapper
+func NewCountingConn(conn net.Conn) net.Conn{
+	c := &connCounting{
+		Conn: conn,
+		closed: false,
+	}
+	runtime.SetFinalizer(c, func(cc *connCounting){
+		if !cc.closed {
+			atomic.AddInt64(&connsCount, -1)
+			cc.closed = true
```
Comment (vitalyisaev2) [inline internal/utils/conn_counter.go]: Isn't it necessary to call `cc.Conn.Close` here?

File: internal/utils/conn_counter.go [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+	return c
+}
+
+func (c *connCounting) Close() error {
+	if !c.closed {
+		atomic.AddInt64(&connsCount, -1)
+		c.closed = true
+	}
+	return c.Conn.Close()
```
Comment (vitalyisaev2) [inline internal/utils/conn_counter.go:81]: 1. looks like you should call `c.Conn.Close()` only within `if` clause
2. don't you need to decrement `createdStacks` counter to print only alive sockets?

File: internal/integration/reql_tests/reql_aggregation_test.go (hunk 1 above)
Comment (CMogilko) [reply internal/integration/reql_tests/reql_aggregation_test.go:1221]: I think this code was once generated from rethinkdb/tests.eml by handwritten templates.

File: internal/utils/conn_counter.go (hunk 3 above)
Comment (CMogilko) [reply internal/utils/conn_counter.go]: it is only debug counter wrapper. If call here, you can't debug leaks

**Pull request merged**